backend/python/app/services/implementations/entity_service.py

```diff
@@ -1,56 +1,57 @@
-from ...models.entity import Entity
-from ...models import db
-from ..interfaces.entity_service import IEntityService
-
-
-class EntityService(IEntityService):
-    def __init__(self, logger):
-        self.logger = logger
-
-    def get_entities(self):
-        # Entity is a SQLAlchemy model, we can use convenient methods provided
-        # by SQLAlchemy like query.all() to query the data
-        return [result.to_dict() for result in Entity.query.all()]
-
-    def get_entity(self, id):
-        # get queries by the primary key, which is id for the Entity table
-        entity = Entity.query.get(id)
-        if entity is None:
-            self.logger.error("Invalid id")
-            raise Exception("Invalid id")
-        return entity.to_dict()
-
-    def create_entity(self, entity):
-        try:
-            new_entity = Entity(**entity.__dict__)
-        except Exception as error:
-            self.logger.error(str(error))
-            raise error
-
-        db.session.add(new_entity)
-        # remember to commit to actually persist into the database
-        db.session.commit()
-
-        return new_entity.to_dict()
-
-    def update_entity(self, id, entity):
-        Entity.query.filter_by(id=id).update(entity.__dict__)
-        updated_entity = Entity.query.get(id)
-        db.session.commit()
-
-        if updated_entity is None:
-            self.logger.error("Invalid id")
-            raise Exception("Invalid id")
-        return updated_entity.to_dict()
-
-    def delete_entity(self, id):
-        deleted = Entity.query.filter_by(id=id).delete()
-        db.session.commit()
-
-        # deleted is the number of rows deleted
-        if deleted == 1:
-            return id
-
-        self.logger.error("Invalid id")
-        raise Exception("Invalid id")
-
\ No newline at end of file
+from ...models.entity import Entity
+from ...models import db
+from ..interfaces.entity_service import IEntityService
+
+from flask import current_app
+
+class EntityService(IEntityService):
+    def __init__(self, logger=current_app.logger):
```

Comment:
git diff looks weird here. This line + flask import are the only changes.

Reply:
Just to provide the background on this, the intention was to limit the flask dependency to only the controller layer - no issues with this though! 🙂